sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/CertificateValidationCheckBase.cs

```diff
@@ -0,0 +1,298 @@
+            }
+        }
+
+        protected void CheckConstructorParameterSyntax(SyntaxNodeAnalysisContext c)
+        {
+            if (c.SemanticModel.GetSymbolInfo(c.Node).Symbol is IMethodSymbol ctor)
+            {
+                AbstractMethodParameterLookup<TArgumentSyntax> methodParamLookup = null;       //Cache, there might be more of them
+                foreach (var param in ctor.Parameters.Where(x => IsValidationDelegateType(x.Type)))
+                {
+                    methodParamLookup = methodParamLookup ?? CreateParameterLookup(c.Node, ctor);
+                    if (methodParamLookup.TryGetSymbolParameter(param, out var argument))
+                    {
+                        TryReportLocations(new InspectionContext(c), ArgumentLocation(argument), ArgumentExpression(argument));
+                    }
+                }
+            }
+        }
+
+        private void TryReportLocations(InspectionContext c, Location primaryLocation, TExpressionSyntax expression)
+        {
+            var locations = ArgumentLocations(c, expression);
+            if (!locations.IsEmpty)
+            {
+                //Report both, assignemnt as well as all implementation occurances
+                c.Context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, primaryLocation, additionalLocations: locations));
+            }
+        }
+
+        private bool IsValidationDelegateType(ITypeSymbol type)
+        {
+            if (type.Is(KnownType.System_Net_Security_RemoteCertificateValidationCallback))
+            {
+                return true;
+            }
+            else if (type is INamedTypeSymbol namedSymbol && type.OriginalDefinition.Is(GenericDelegateType())) 
+            {
+                //HttpClientHandler.ServerCertificateCustomValidationCallback uses Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool>
+                //We're actually looking for Func<Any Sender, X509Certificate2, X509Chain, SslPolicyErrors, bool>
+                var parameters = namedSymbol.DelegateInvokeMethod.Parameters;
+                return parameters.Length == 4   //And it should! T1, T2, T3, T4
+                    && parameters[0].Type.IsClassOrStruct() //We don't care about common (Object) nor specific (HttpRequestMessage) type of Sender
+                    && parameters[1].Type.Is(KnownType.System_Security_Cryptography_X509Certificates_X509Certificate2)
+                    && parameters[2].Type.Is(KnownType.System_Security_Cryptography_X509Certificates_X509Chain)
+                    && parameters[3].Type.Is(KnownType.System_Net_Security_SslPolicyErrors)
+                    && namedSymbol.DelegateInvokeMethod.ReturnType.Is(KnownType.System_Boolean);
+            }
+            return false;
+        }
+
+        private ImmutableArray<Location> ArgumentLocations(InspectionContext c, TExpressionSyntax expression)
+        {
+            switch (TryExtractAddressOfOperand(expression))
```

Comment:
I suggest to rename `TryExtractAddressOfOperand` as it is very VB.NET specific. Something like `GetMethodReferenceNode` (name subject to change) would make sense in both C# & VB contexts